### DISCUSSION TITLE
touch typing + titling wm + vim + pop shell = my gnome keybindings 😃

### DISCUSSION THREAD

**Discussion by cutbypham** [pop-os/shell#1178] [Show and tell]
> gnome keybindings
```
[/]
close=['<Alt>q']
cycle-windows=['<Alt>Tab']
cycle-windows-backward=['<Shift><Alt>Tab']
maximize=@as []
minimize=['<Alt>comma']
move-to-monitor-down=@as []
move-to-monitor-left=@as []
move-to-monitor-right=@as []
move-to-monitor-up=@as []
move-to-workspace-0=['<Shift><Alt>parenright']
move-to-workspace-1=['<Shift><Alt>exclam']
move-to-workspace-2=['<Shift><Alt>at']
move-to-workspace-3=['<Shift><Alt>numbersign']
move-to-workspace-4=['<Shift><Alt>dollar']
move-to-workspace-5=['<Shift><Alt>percent']
move-to-workspace-6=['<Shift><Alt>and']
move-to-workspace-7=['<Shift><Alt>ampersand']
move-to-workspace-8=['<Shift><Alt>asterisk']
move-to-workspace-9=['<Shift><Alt>parenleft']
move-to-workspace-down=@as []
move-to-workspace-left=['<Shift><Alt>h']
move-to-workspace-right=['<Shift><Alt>l']
move-to-workspace-up=@as []
show-desktop=['<Alt>d']
switch-applications=@as []
switch-applications-backward=@as []
switch-panels=@as []
switch-panels-backward=@as []
switch-to-workspace-0=['<Alt>0']
switch-to-workspace-1=['<Alt>1']
switch-to-workspace-10=['<Alt>0']
switch-to-workspace-2=['<Alt>2']
switch-to-workspace-3=['<Alt>3']
switch-to-workspace-4=['<Alt>4']
switch-to-workspace-5=['<Alt>5']
switch-to-workspace-6=['<Alt>6']
switch-to-workspace-7=['<Alt>7']
switch-to-workspace-8=['<Alt>8']
switch-to-workspace-9=['<Alt>9']
switch-to-workspace-down=['<Primary><Super>Down', '<Primary><Super>j']
switch-to-workspace-left=['<Alt>h']
switch-to-workspace-right=['<Alt>l']
switch-to-workspace-up=['<Primary><Super>Up', '<Primary><Super>k']
toggle-maximized=['<Alt>z']
unmaximize=@as []
```
redit: <https://github.com/thuanpham2311/dotfiles>

> If you want to try this, I recommend you create a new user, and run `dconf load /org/gnome/desktop/wm/keybindings/ < file_path_this_script`

**My laptop**
![image](https://user-images.githubusercontent.com/40050527/133248117-6431f5e0-2c02-43c1-8ea8-e9b28b00e705.png)

**My keyboard layout**
![image](https://user-images.githubusercontent.com/40050527/133246923-686862ec-7c06-473c-9975-2fd66c556e6d.png)

😁 Feel free to discuss with me.

**Comment by mmstick** [comment]
Using Alt for OS shortcuts is a bad idea because Super is reserved for OS shortcuts, and Alt is reserved for applications. So by defining OS shortcuts to use Alt, keyboard shortcuts in applications that use the same shortcuts will stop working.

Firefox and VS Code, for example, uses Alt + Num for switching between tabs. Toolbars and mnemonic buttons respond to Alt + Char. Such as Alt + B bringing up the bookmark tooltip in Firefox

**Reply by cutbypham** [reply]
- Yeah, that is the price when I use `Alt`.
- For me, `alt` easy to press for my keyboard layout.
![image](https://user-images.githubusercontent.com/40050527/133257636-ca2b9fea-03fc-4af7-98ee-2f564b80442a.png)
- I daily use terminal for coding, brave (chromium base) in my default browser, etc. It doesn't make me feel crazy 😅 with `Alt`.

**Reply by cutbypham** [reply]
Tips
- My solution in firefox,
![image](https://user-images.githubusercontent.com/40050527/133258698-a7451e52-ac56-46a4-99d6-e0e7597d6604.png)
- I use extention [surfingkeys](https://github.com/brookhong/Surfingkeys), you will like it if you vim user

**Reply by cutbypham** [reply]
A other solution if you don't like to use `Alt`, and have a same keyboard layout with me, you can remap `Alt` key to `super/mod` key, 😃

**Reply by cutbypham** [reply]
And other solution, you can buy system 76 keyboard 😁
![image](https://user-images.githubusercontent.com/40050527/133260383-9d553d31-c006-4434-b49c-6abcc44fcf8d.png)

**Comment by michaelmob** [comment]
Here's how to easily swap Alt and Win

![image](https://user-images.githubusercontent.com/4530010/134445141-0b26904b-ff3b-4ff2-9508-8ee622badd4e.png)

**Reply by cutbypham** [reply]
okay, let rewrite my keyboard config :smile:

**Reply by cutbypham** [reply]
I don't know why, it don't work with my config, I try with option below.

Hm hello my old `Alt` friend 😅, i will come back with you again 😁

![image](https://user-images.githubusercontent.com/40050527/134448912-5d2cd389-7f36-4366-a7cf-6b530a6e2ed9.png)

**Comment by cutbypham** [comment]
```json
{
  "/org/gnome/mutter/keybindings/": "[/]toggle-tiled-left=@as []\ntoggle-tiled-right=@as []",
  "/org/gnome/mutter/wayland/keybindings/": "[/]\nrestore-shortcuts=@as []",
  "/org/gnome/shell/keybindings/": "[/]\nopen-application-menu=@as []\nswitch-to-application-1=['']\nswitch-to-application-2=['']\nswitch-to-application-3=['']\nswitch-to-application-4=['']\nswitch-to-application-5=['']\nswitch-to-application-6=['']\nswitch-to-application-7=['']\nswitch-to-application-8=['']\nswitch-to-application-9=['']\ntoggle-message-tray=['<Super>v']\ntoggle-overview=@as []",
  "/org/gnome/desktop/wm/keybindings/": "[/]\nclose=['<Super>q', '<Alt>F4']\nmaximize=@as []\nminimize=['<Super>comma']\nmove-to-monitor-down=@as []\nmove-to-monitor-left=@as []\nmove-to-monitor-right=@as []\nmove-to-monitor-up=@as []\nmove-to-workspace-1=['<Super><Shift>1']\nmove-to-workspace-10=['<Super><Shift>0']\nmove-to-workspace-2=['<Super><Shift>2']\nmove-to-workspace-3=['<Super><Shift>3']\nmove-to-workspace-4=['<Super><Shift>4']\nmove-to-workspace-5=['<Super><Shift>5']\nmove-to-workspace-6=['<Super><Shift>6']\nmove-to-workspace-7=['<Super><Shift>7']\nmove-to-workspace-8=['<Super><Shift>8']\nmove-to-workspace-9=['<Super><Shift>9']\nmove-to-workspace-down=@as []\nmove-to-workspace-up=@as []\nswitch-to-workspace-1=['<Super>1']\nswitch-to-workspace-10=['<Super>0']\nswitch-to-workspace-2=['<Super>2']\nswitch-to-workspace-3=['<Super>3']\nswitch-to-workspace-4=['<Super>4']\nswitch-to-workspace-5=['<Super>5']\nswitch-to-workspace-6=['<Super>6']\nswitch-to-workspace-7=['<Super>7']\nswitch-to-workspace-8=['<Super>8']\nswitch-to-workspace-9=['<Super>9']\nswitch-to-workspace-down=['<Primary><Super>Down', '<Primary><Super>j']\nswitch-to-workspace-left=@as []\nswitch-to-workspace-right=@as []\nswitch-to-workspace-up=['<Primary><Super>Up', '<Primary><Super>k']\ntoggle-maximized=['<Super>m']\nunmaximize=@as []"
}
```
